app/dashboard/models.py

```diff
@@ -2226,6 +2226,26 @@ def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
 
         return params
 
+    @property
+    def locations(self):
```

Comment:
i wonder what the performance profile of this method is like when it's called repeatedly 🤔 IIRC -- that was a major problem when i had worked on the PR...

i wonder if it makes sense to cache the location information on a profile-by-profile basis... that way we just incur a DB hit when we look it up instead of calling `get_location_from_ip()` which seems to be pretty performance reducing..